File: Template Tools/PV_Tool/dwg_import.py
import pandas as pd
import sys
import excel_handler
import file_finder
import create_folder
import os

# initialize variables and lists
command_input = sys.argv
excel_file_path = command_input[1]
workbook_folder = command_input[2]
xl_data_list = []
xl_named_data_list = []
xl_dict = {}
xl_sheet_name = "All Data Outputs"
pv_tool_loc = f'C:/Template Tools/PV_tool'
pv_tool_dwg_loc = f'{pv_tool_loc}/dwgs'
sld_dwg_loc = f'{pv_tool_dwg_loc}/line_diagram'

# xrefs
xref_folder_path = f'{workbook_folder}/xref'
company_logo = f'company_logo.dwg'
mounting_detail = f'mounting_detail.dwg'
inv_strings = f'inv_strings.dwg'
battery = f'batt.dwg'
ac_disco_meter_combo = f'combo.dwg'
interconnection = f'interconnections.dwg'
meter = f'meter.dwg'
other_sld = f'other_sld.dwg'
labels = f'labels.dwg'

"""
# test values
mounting_detail = f'mounting_detail.txt'
company_logo = f'company_logo.txt'
inv_strings = f'inv_strings.txt'
ac_disco_meter = f'ac_disco_pv_meter.txt'
interconnection = f'interconnection.txt'
"""

dwg_list = [
    company_logo,
    mounting_detail,
    inv_strings,
    battery,
    ac_disco_meter_combo,
    interconnection,
    meter,
    other_sld,
    labels
        ]

# import excel workbook as a list, xl_data_list
# convert xl_data_list into dataframe
# rewrite dataframe as a dictionary
excel_handler.import_sheet(excel_file_path, xl_sheet_name, xl_data_list)
xl_named_data_list = pd.DataFrame(xl_data_list[1:], columns=xl_data_list[0])

for i in range(0, len(xl_data_list) - 1):
    xl_dict[xl_named_data_list.iloc[i, 0]] = xl_named_data_list.iloc[i, 1]

# xref folder names
company_folder = xl_dict['xl_company']
mounting_detail_folder = xl_dict['xl_mounting_detail']
inv_strings_folder = xl_dict['xl_inv_strings']      # this can be calculated by system calcs instead of excel
battery_folder = xl_dict['xl_battery']
combo_folder = xl_dict['xl_combo']
interconnection_folder = xl_dict['xl_interconnection']
meter_folder = xl_dict['xl_meter']
other_sld_folder = xl_dict['xl_other_sld']
labels_folder = xl_dict['xl_labels']

# xref paths
logo_path = f'{pv_tool_dwg_loc}/company_logos/{company_folder}/{company_logo}'
mounting_detail_path = f'{pv_tool_dwg_loc}/mounting_detail/{mounting_detail_folder}/{mounting_detail}'
inv_str_path = f'{sld_dwg_loc}/strings/{inv_strings_folder}/{inv_strings}'
battery_path = f'{sld_dwg_loc}/batteries/{battery_folder}/{battery}'
combo_path = f'{sld_dwg_loc}/combos/{combo_folder}/{ac_disco_meter_combo}'
interconnection_path = f'{sld_dwg_loc}/interconnections/{interconnection_folder}/{interconnection}'
meter_path = f'{sld_dwg_loc}/meter/{meter_folder}/{meter}'
other_sld_path = f'{sld_dwg_loc}/other/{other_sld_folder}/{other_sld}'
labels_path = f'{pv_tool_dwg_loc}/labels/{labels_folder}/{labels}'


dwg_path_list = [
    logo_path,
    mounting_detail_path,
    inv_str_path,
    battery_path,
    combo_path,
    interconnection_path,
    meter_path,
    other_sld_path,
    labels_path
        ]

# copy xrefs from pv_tool into project folder
file_finder.copy_file(dwg_path_list, xref_folder_path, dwg_list)

# TO DO
# System calcs pass inv_str value
#

File: tools/dwgImport.js
const FS = require('fs');
const PATH = require('path');
module.exports = {
    // takes json object and destination folder
    // copies specific items mentioned in json object to dest folder
    moveDwgs: (json, dest) => {

        const fromFolder = `${__dirname}/../dwgs`;
        dest = `${__dirname}/../${dest}`;

        // file name constants
        const company_logo = 'company_logo.dwg';
        const mounting_detail = 'mounting_detail.dwg';
        const inv_strings = 'inv_strings.dwg';
        const battery = 'batt.dwg';
        const ac_disco_meter_combo = 'combo.dwg';
        const interconnection = 'interconnections.dwg';
        const meter = 'meter_boi.dwg';
        const other_sld = 'other_sld.dwg';
        const labels = 'labels.dwg';


        // get paths from json obj
        const company_folder = json['xl_company'];
        const mounting_detail_folder = json['xl_mounting_detail'];
        const inv_strings_folder = json['xl_inv_strings'];
        const battery_folder = json['xl_battery'];
        const combo_folder = json['xl_combo'];
        const interconnection_folder = json['xl_interconnection'];
        const meter_folder = json['xl_meter'];
        const other_sld_folder = json['xl_other_sld'];
        const labels_folder = json['xl_labels'];

        // xref paths
        const logo_path = `${fromFolder}/company_logos/${company_folder}/${company_logo}`
        const mounting_detail_path = `${fromFolder}/mounting_detail/${mounting_detail_folder}/${mounting_detail}`
        const inv_str_path = `${fromFolder}/line_diagram/strings/${inv_strings_folder}/${inv_strings}`
        const battery_path = `${fromFolder}/line_diagram/batteries/${battery_folder}/${battery}`
        const combo_path = `${fromFolder}/line_diagram/combos/${combo_folder}/${ac_disco_meter_combo}`
        const interconnection_path = `${fromFolder}/line_diagram/interconnections/${interconnection_folder}/${interconnection}`
        const meter_path = `${fromFolder}/line_diagram/meter/${meter_folder}/${meter}`
        const other_sld_path = `${fromFolder}/line_diagram/other/${other_sld_folder}/${other_sld}`
        const labels_path = `${fromFolder}/labels/${labels_folder}/${labels}`

        const dwg_path_list = [
            logo_path,
            mounting_detail_path,
            inv_str_path,
            battery_path,
            combo_path,
            interconnection_path,
            meter_path,
            other_sld_path,
            labels_path
        ];

        copyFiles(dwg_path_list, dest)
    }
}

function copyFiles(dwg_path_list, dest) {
    if (FS.existsSync(dest)) {
        for (const file of dwg_path_list) {
            if (FS.existsSync(file)) {
                FS.copyFileSync(file, `${dest}/${PATH.parse(file).base}`)
            }
        }
    }
}